src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -1842,6 +1845,62 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// 383490 - Wildfire
+class spell_mage_wildfire : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MAGE_WILDFIRE_CASTER_CRIT })
+            && ValidateSpellEffect({ { SPELL_MAGE_WILDFIRE_TALENT, EFFECT_2 } });
+    }
+
+    void HandleProc(AuraEffect const* aurEff, ProcEventInfo const& /*eventInfo*/)
+    {
+        PreventDefaultAction();
+
+        AuraEffect const* wildfireCrit = GetCaster()->GetAuraEffect(SPELL_MAGE_WILDFIRE_TALENT, EFFECT_2);
+
+        GetCaster()->CastSpell(GetTarget(), SPELL_MAGE_WILDFIRE_CASTER_CRIT, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringAura = aurEff,
+            .SpellValueOverrides = { { SPELLVALUE_BASE_POINT0, wildfireCrit->GetAmount() } }
+        });
+    }
+
+    void Register() override
+    {
+        OnEffectProc += AuraEffectProcFn(spell_mage_wildfire::HandleProc, EFFECT_0, SPELL_AURA_PROC_TRIGGER_SPELL);
+    }
+};
+
+// 383492 - Wildfire
+class spell_mage_wildfire_area : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MAGE_WILDFIRE_AREA_CRIT })
+            && ValidateSpellEffect({ { SPELL_MAGE_WILDFIRE_TALENT, EFFECT_3 } });
+    }
+
+    void AreaCrit(SpellEffIndex effIndex)
+    {
+        PreventHitDefaultEffect(effIndex);
+
+        AuraEffect const* wildfireAreaCrit = GetCaster()->GetAuraEffect(SPELL_MAGE_WILDFIRE_TALENT, EFFECT_3);
```

Comment:
missing nullcheck, may crash if i cast this spell without having the talent (or check if you have the talent in `Load`